docs(rbac): Document how to mount the RBAC policy CSV file using the Operator [RHIDP-1783] (#499)
redhat-developer/red-hat-developers-documentation-rhdh: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc

[id='proc-mounting-the-policy-csv-file-using-the-operator_{context}']
= Mounting `policy.csv` file using the {product-short} Operator

> You can add your `policy.csv` file using the {product-short} Operator by creating a `ConfigMap` and mounting it through your Custom Resource.

Reviewer: ```suggestion
When the {product} is deployed with the Operator, you can add your `policy.csv` file using the {product-short} Operator by creating a `ConfigMap` and mounting it through your Custom Resource (CR).
```

//replace with a link to the installation guide.

.Procedure

. In {ocp-short}, create a ConfigMap to hold the policies as shown in the following example:
+
--
.Example `ConfigMap`
[source,yaml]
----
kind: ConfigMap
apiVersion: v1
metadata:
  name: rbac-policy
data:
  rbac-policy.csv: |
    p, role:default/guests, catalog-entity, read, allow
    p, role:default/guests, catalog.entity.create, create, allow

    g, user:default/<YOUR_USER>, role:default/guests
----
--

. Update the policy path in your custom `app-config.yaml` ConfigMap as follows:
+
--
.Example `app-config.yaml` file
[source,yaml]
----
permission:
  enabled: true
  rbac:
    policies-csv-file: ./rbac-policy.csv
----
--

. From the *Developer* perspective in the {ocp-short} web console, select the *Topology* view.
. Click the overflow menu for the {product} instance that you want to use and select *Edit Backstage* to load the YAML view of the {product} instance.
. In the CR, enter the name of the custom `rbac-policy` ConfigMap as the value for the `spec.application.extraFiles.configMaps` field. For example:
+

> [source, yaml]

Reviewer: ```suggestion
.Example custom resource
[source, yaml]
```

----
apiVersion: v1
kind: ConfigMap
metadata:
  name: example
spec:
  application:
    appConfig:
      mountPath: /opt/app-root/src
      configMaps:
        - name: app-config-rhdh
    extraEnvs:
      secrets:
        - name: secrets-rhdh
    extraFiles:
      mountPath: /opt/app-root/src
      configMaps:
        - name: rbac-policy
    replicas: 1
    route:
      enabled: true
  database:
    enableLocalDb: true
----
. Click *Save*.
. Navigate back to the *Topology* view and wait for the {product} pod to start.

> . Click the *Open URL* icon to use the {product} platform with the configuration changes.

Reviewer: ```suggestion
. Click the *Open URL* icon to access the {product} platform with the updated configuration settings.
```

[id='proc-mounting-the-policy-csv-file-using-the-operator_{context}']
= Mounting `policy.csv` file using the {product-short} Operator

When the {product} is deployed with the Operator, you can add your `policy.csv` file using the {product-short} Operator by creating a `ConfigMap` and mounting it through your Custom Resource (CR).

.Prerequisites

* You are logged in to your {ocp-short} account using the {ocp-short} web console.
* {product} is installed and deployed using the Operator.
* You have added a custom configuration file to {ocp-short}. For more information, see link:{LinkAdminGuide}[Adding a custom configuration file to {ocp-short}].
+
//For more information about installing the {product} on {ocp-short} using the Operator, see xref:proc-install-rhdh-ocp-operator_{context}[].

> //replace with a link to the installation guide.

Reviewer 2: Should this be here?

Reviewer: We already have a Jira for replacing the links, and another writer is currently addressing all the links in a single PR. Therefore, we can leave them as they are in this PR: https://issues.redhat.com/browse/RHIDP-3377.

    g, user:default/<YOUR_USER>, role:default/guests
----
--

. Update the policy path in your custom `app-config.yaml` ConfigMap as follows:
+
--
.Example `app-config.yaml` file
[source,yaml]
----
permission:
  enabled: true
  rbac:
    policies-csv-file: ./rbac-policy.csv
----
--

. From the *Developer* perspective in the {ocp-short} web console, select the *Topology* view.
. Click the overflow menu for the {product} instance that you want to use and select *Edit Backstage* to load the YAML view of the {product} instance.
. In the CR, enter the name of the custom `rbac-policy` ConfigMap as the value for the `spec.application.extraFiles.configMaps` field. For example:
+
.Example custom resource
[source, yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: example
spec:
  application:
    appConfig:
      mountPath: /opt/app-root/src
      configMaps:
        - name: app-config-rhdh
    extraEnvs:
      secrets:
        - name: secrets-rhdh
    extraFiles:
      mountPath: /opt/app-root/src
      configMaps:
        - name: rbac-policy
    replicas: 1
    route:
      enabled: true
  database:
    enableLocalDb: true
----
. Click *Save*.

> . Navigate back to the *Topology* view and wait for the {product} pod to start.

Reviewer 3: For clarity, consider isolating the verification steps:

```suggestion
.Verification

. Navigate back to the *Topology* view and wait for the {product} pod to start.
```
